langserver/build_context.go

```diff
@@ -82,8 +82,11 @@ func fsBuildContext(ctx context.Context, orig *build.Context, fs ctxvfs.FileSyst
 // * if the file is in the xtest package (package p_test not package p),
 //   it returns build.Package only representing that xtest package
 func ContainingPackage(bctx *build.Context, filename string) (*build.Package, error) {
-	if !filepath.IsAbs(bctx.GOPATH) || strings.Contains(bctx.GOPATH, string(os.PathListSeparator)) {
-		return nil, fmt.Errorf("build context GOPATH must contain exactly 1 entry and it must be an absolute path (GOPATH=%q)", bctx.GOPATH)
+	gopaths := strings.Split(bctx.GOPATH, string(os.PathListSeparator))
```

Comment:
There's also https://godoc.org/go/build#Context.SrcDirs which is slightly higher level. It omits non-existing directories, and includes GOROOT.